src/Ilios/WebBundle/Controller/IcsController.php

```diff
@@ -21,34 +20,42 @@ class IcsController extends Controller
 
     public function indexAction(Request $request, $key)
     {
-        $userManager = $this->container->get('ilioscore.user.manager');
-        $user = $userManager->findUserBy(array('icsFeedKey' => $key));
-        
+        $userHandler = $this->container->get('ilioscore.user.handler');
+        $user = $userHandler->findUserBy(array('icsFeedKey' => $key));
+
         if (!$user) {
             throw new NotFoundHttpException();
         }
-        
+
         $calendar = new ICS\Calendar('Ilios Calendar for ' . $user->getFirstAndLastName());
         $calendar->setPublishedTTL('P1H');
         
         $from = new \DateTime(self::LOOK_BACK);
         $to =  new \DateTime(self::LOOK_FORWARD);
 
-        $events = $userManager->findEventsForUser(
-            $user->getId(),
-            $from,
-            $to
-        );
+        $events = $userHandler->addInstructorsToEvents($userHandler->findEventsForUser($user->getId(), $from, $to));
+
+        //Only show published events
+        $events = array_filter($events, function ($event) {
```

Comment:
i suggest moving this up. filter out un-published events first before loading/attaching additional data points, such as instructors.